DataFormats/PatCandidates/interface/PackedCandidate.h

```diff
@@ -204,12 +204,12 @@ namespace pat {
           // Need to trigger unpacking in iOther
           p4_(new PolarLorentzVector(iOther.polarP4())),
           p4c_(new LorentzVector(iOther.p4())),
-          vertex_(new Point(iOther.vertex())),
-          dxy_(iOther.dxy_),
-          dz_(iOther.dz_),
-          dphi_(iOther.dphi_),
-          deta_(iOther.deta_),
-          dtrkpt_(iOther.dtrkpt_),
+          vertex_((iOther.vertex_ ? new Point(iOther.vertex()) : nullptr)),
+          dxy_(vertex_ ? iOther.dxy_ : 0),
+          dz_(vertex_ ? iOther.dz_ : 0),
+          dphi_(vertex_ ? iOther.dphi_ : 0),
+          deta_(vertex_ ? iOther.deta_ : 0),
+          dtrkpt_(vertex_ ? iOther.dtrkpt_ : 0),
```

Comment:
this doesn't look right; having a null vertex does not imply zero values in these parameters
https://github.com/cms-sw/cmssw/blob/22f5521804d2e9a1ffee48aeefecca1bef226295/DataFormats/PatCandidates/src/PackedCandidate.cc#L122-L132
I think that the unpacking of parameters not dependent on the value of the `vertexRef` should still happen.

Reply:
uhm, on a second thought, perhaps not and the changes are OK, but that becomes a bit more obvious only after seeing that all cases to access the `dxy_` through `dtrkpt_` values rely on having `unpackVtx` called.

I see a path after this PR change that a candidate that comes out of this `PackedCandidate(const PackedCandidate &iOther)`  with `iOther.vertex_ == nullptr` can get to e.g. a call to `::packVtx` without a preceding `unpackVtx` (e.g. via `::setTrackProperties`) and lead to a crash in 
https://github.com/cms-sw/cmssw/blob/22f5521804d2e9a1ffee48aeefecca1bef226295/DataFormats/PatCandidates/src/PackedCandidate.cc#L26-L29

This looks like a `maybeUnpackBoth()` call is needed in `::setTrackProperties`

Reply:
Good catch! I made the change.